tests/release/pipelines/rhtap_service_push.go

```diff
@@ -0,0 +1,352 @@
+	defaultEcPolicySpec := ecp.EnterpriseContractPolicySpec{
+		Description: "Red Hat's enterprise requirements",
+		PublicKey:   "k8s://openshift-pipelines/public-key",
+		Sources: []ecp.Source{{
+			Name:   "Default",
+			Policy: []string{releasecommon.EcPolicyLibPath, releasecommon.EcPolicyReleasePath},
+			Data:   []string{releasecommon.EcPolicyDataBundle, releasecommon.EcPolicyDataPath},
+		}},
+		Configuration: &ecp.EnterpriseContractPolicyConfiguration{
+			Exclude: []string{"step_image_registries", "tasks.required_tasks_found:prefetch-dependencies"},
+			Include: []string{"@slsa3"},
+		},
+	}
+
+	_, err := managedFw.AsKubeDeveloper.TektonController.CreateEnterpriseContractPolicy(rhtapECPName, managedNamespace, defaultEcPolicySpec)
+	Expect(err).NotTo(HaveOccurred())
+
+}
+
+func createRHTAPReleasePlanAdmission(rhtapRPAName string, managedFw framework.Framework, devNamespace, managedNamespace, rhtapAppName, rhtapECPName, pathInRepoValue string) {
+	var err error
+
+	data, err := json.Marshal(map[string]interface{}{
+		"mapping": map[string]interface{}{
+			"components": []map[string]interface{}{
+				{
+					"name":       rhtapComponent.GetName(),
+					"repository": "quay.io/redhat-pending/rhtap----konflux-release-e2e",
+					"tags": []string{"latest", "latest-{{ timestamp }}", "testtag",
+						"testtag-{{ timestamp }}", "testtag2", "testtag2-{{ timestamp }}"},
+					"source": map[string]interface{}{
+						"git": map[string]interface{}{
+							"url": rhtapGitSourceURL,
+						},
+					},
+				},
+			},
+		},
+		"pyxis": map[string]interface{}{
+			"server": "stage",
+			"secret": "pyxis",
+		},
+		"targetGHRepo": "hacbs-release/infra-deployments",
+		"githubAppID": "932323",
+		"githubAppInstallationID" : "52284535",
+		"infra-deployment-update-script": "sed -i -e 's|\\(https://github.com/hacbs-release/release-service/config/default?ref=\\)\\(.*\\)|\\1{{ revision }}|' -e 's/\\(newTag: \\).*/\\1{{ revision }}/' components/release/development/kustomization.yaml",
+		"sign": map[string]interface{}{
+			"configMapName": "hacbs-signing-pipeline-config-redhatbeta2",
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	_, err = managedFw.AsKubeAdmin.ReleaseController.CreateReleasePlanAdmission(rhtapRPAName, managedNamespace, "", devNamespace, rhtapECPName, rhtapServiceAccountName, []string{rhtapAppName}, true, &tektonutils.PipelineRef{
+		Resolver: "git",
+		Params: []tektonutils.Param{
+			{Name: "url", Value: releasecommon.RelSvcCatalogURL},
+			{Name: "revision", Value: releasecommon.RelSvcCatalogRevision},
+			{Name: "pathInRepo", Value: pathInRepoValue},
+		},
+	}, &runtime.RawExtension{
+		Raw: data,
+	})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// prepareMergedPR function is to prepare a merged PR in source repo for testing update-infra-deployments task
+func prepareMergedPR(devFw *framework.Framework, testBaseBranchName, testPRBranchName string) (int, string) {
```

Comment:
It isn't clear to me why we need this function. We shouldn't have to trigger things by merging a PR, we can just do it with a component build, no?

Reply:
It's for verifying the result of "update-infra-deployment" task after the rhtap-service-push is done.  We usually check if image reference is updated to the right one. But the checking may fail for race condition when the test is triggered by the CI jobs from several PRs at almost the same time.   This function is to create one PR for each test run. Then the result of each test can be verified by checking if the PR is added to the infra-deployment's PR.